File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Classes/StartingPiece.java
package com.alphadevelopmentsolutions.frcscout.Classes;

/**
 * Pieces available during a match
 */
public enum StartingPiece
{
    HATCH,
    CARGO;

    /**
     * Converts a string value to enum
     * @param piece string enum name
     * @return enum converted from string
     */
    public static StartingPiece getPieceFromString(String piece)
    {
        if(piece.toLowerCase().equals(StartingPiece.HATCH.name().toLowerCase()))
            return StartingPiece.HATCH;
        else
            return StartingPiece.CARGO;
    }
}

File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Enums/StartingPiece.kt
package com.alphadevelopmentsolutions.frcscout.Enums

/**
 * Pieces available during a match
 */
enum class StartingPiece
{
    HATCH,
    CARGO;


    companion object
    {

        /**
         * Converts a string value to enum
         * @param piece string enum name
         * @return enum converted from string
         */
        fun getPieceFromString(piece: String): StartingPiece
        {
            return if (piece.toLowerCase() == StartingPiece.HATCH.name.toLowerCase())
                HATCH
            else
                CARGO
        }
    }
}
